### ISSUE TITLE
visual studio code: debian 11 and older apt/dpkg do not support tar.zst

### ISSUE THREAD

**Issue by oziljak666** [Botspot/pi-apps#2655]
### Confirmations

- [X] I searched for similar bug reports in https://github.com/Botspot/pi-apps/issues and found none were relevant.
- [X] I tried updating Pi-Apps and the issue is still not fixed. [How to update Pi-Apps?](https://pi-apps.io/wiki/getting-started/updating-pi-apps/)
- [X] IMPORTANT: I promise to respond to comments on this issue and write a good bug report https://www.softwaretestinghelp.com/how-to-write-good-bug-report/

### What happened?

app wont install, i tried uninstall and again install

### Description
```
Will install these packages: https://aka.ms/linux-arm64-deb 
Waiting until APT locks are released... Done
/tmp/linux-arm64-deb is not ending with .deb, renaming it to '/tmp/linux-arm64-deb.deb'...
Downloading linux-arm64-deb to /tmp/linux-arm64-deb.deb... 
--2024-10-08 14:58:46--  https://aka.ms/linux-arm64-deb
Resolving aka.ms (aka.ms)... 72.246.170.70
Connecting to aka.ms (aka.ms)|72.246.170.70|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://update.code.visualstudio.com/latest/linux-deb-arm64/stable [following]
--2024-10-08 14:58:47--  https://update.code.visualstudio.com/latest/linux-deb-arm64/stable
Resolving update.code.visualstudio.com (update.code.visualstudio.com)... 13.107.246.45, 2620:1ec:bdf::45
Connecting to update.code.visualstudio.com (update.code.visualstudio.com)|13.107.246.45|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://vscode.download.prss.microsoft.com/dbazure/download/stable/d78a74bcdfad14d5d3b1b782f87255d802b57511/code_1.94.0-1727877189_arm64.deb [following]
--2024-10-08 14:58:47--  https://vscode.download.prss.microsoft.com/dbazure/download/stable/d78a74bcdfad14d5d3b1b782f87255d802b57511/code_1.94.0-1727877189_arm64.deb
Resolving vscode.download.prss.microsoft.com (vscode.download.prss.microsoft.com)... 151.101.242.172, 2a04:4e42:39::684
Connecting to vscode.download.prss.microsoft.com (vscode.download.prss.microsoft.com)|151.101.242.172|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 105335974 (100M) [application/octet-stream]
Saving to: ‘/tmp/linux-arm64-deb.deb’


/tmp/linux-arm64-de   0%[                    ]       0  --.-KB/s               
/tmp/linux-arm64-de   0%[                    ] 895.21K  3.83MB/s               
/tmp/linux-arm64-de   2%[                    ]   2.78M  6.48MB/s               
/tmp/linux-arm64-de   4%[                    ]   4.69M  7.45MB/s               
/tmp/linux-arm64-de   6%[>                   ]   6.23M  7.51MB/s               
/tmp/linux-arm64-de   7%[>                   ]   7.82M  7.59MB/s               
/tmp/linux-arm64-de   9%[>                   ]   9.44M  7.67MB/s               
/tmp/linux-arm64-de  11%[=>                  ]  11.11M  7.75MB/s               
/tmp/linux-arm64-de  12%[=>                  ]  12.66M  7.75MB/s               
/tmp/linux-arm64-de  14%[=>                  ]  14.38M  7.84MB/s               
/tmp/linux-arm64-de  16%[==>                 ]  16.09M  7.91MB/s               
/tmp/linux-arm64-de  17%[==>                 ]  17.81M  7.97MB/s               
/tmp/linux-arm64-de  19%[==>                 ]  19.53M  8.02MB/s               
/tmp/linux-arm64-de  21%[===>                ]  21.31M  8.08MB/s               
/tmp/linux-arm64-de  22%[===>                ]  23.09M  8.12MB/s               
/tmp/linux-arm64-de  24%[===>                ]  24.92M  8.18MB/s    eta 9s     
/tmp/linux-arm64-de  26%[====>               ]  26.70M  8.47MB/s    eta 9s     
/tmp/linux-arm64-de  28%[====>               ]  28.33M  8.63MB/s    eta 9s     
/tmp/linux-arm64-de  29%[====>               ]  29.40M  8.16MB/s    eta 9s     
/tmp/linux-arm64-de  31%[=====>              ]  31.28M  8.28MB/s    eta 9s     
/tmp/linux-arm64-de  32%[=====>              ]  33.10M  8.34MB/s    eta 8s     
/tmp/linux-arm64-de  34%[=====>              ]  34.52M  8.30MB/s    eta 8s     
/tmp/linux-arm64-de  36%[======>             ]  36.34M  8.35MB/s    eta 8s     
/tmp/linux-arm64-de  37%[======>             ]  38.17M  8.41MB/s    eta 8s     
/tmp/linux-arm64-de  39%[======>             ]  39.97M  8.49MB/s    eta 8s     
/tmp/linux-arm64-de  41%[=======>            ]  41.79M  8.52MB/s    eta 7s     
/tmp/linux-arm64-de  43%[=======>            ]  43.59M  8.54MB/s    eta 7s     
/tmp/linux-arm64-de  45%[========>           ]  45.51M  8.60MB/s    eta 7s     
/tmp/linux-arm64-de  47%[========>           ]  47.46M  8.66MB/s    eta 7s     
/tmp/linux-arm64-de  49%[========>           ]  49.37M  8.71MB/s    eta 7s     
/tmp/linux-arm64-de  51%[=========>          ]  51.27M  8.74MB/s    eta 6s     
/tmp/linux-arm64-de  52%[=========>          ]  53.19M  8.79MB/s    eta 6s     
/tmp/linux-arm64-de  54%[=========>          ]  55.10M  8.83MB/s    eta 6s     
/tmp/linux-arm64-de  56%[==========>         ]  57.03M  9.27MB/s    eta 6s     
/tmp/linux-arm64-de  58%[==========>         ]  58.77M  9.08MB/s    eta 6s     
/tmp/linux-arm64-de  59%[==========>         ]  60.19M  8.98MB/s    eta 5s     
/tmp/linux-arm64-de  61%[===========>        ]  61.65M  8.98MB/s    eta 5s     
/tmp/linux-arm64-de  62%[===========>        ]  62.94M  8.83MB/s    eta 5s     
/tmp/linux-arm64-de  63%[===========>        ]  64.05M  8.61MB/s    eta 5s     
/tmp/linux-arm64-de  64%[===========>        ]  65.20M  8.37MB/s    eta 5s     
/tmp/linux-arm64-de  66%[============>       ]  66.37M  8.18MB/s    eta 4s     
/tmp/linux-arm64-de  67%[============>       ]  67.54M  7.97MB/s    eta 4s     
/tmp/linux-arm64-de  68%[============>       ]  68.77M  7.73MB/s    eta 4s     
/tmp/linux-arm64-de  69%[============>       ]  69.99M  7.54MB/s    eta 4s     
/tmp/linux-arm64-de  70%[=============>      ]  71.23M  7.36MB/s    eta 4s     
/tmp/linux-arm64-de  72%[=============>      ]  72.47M  7.06MB/s    eta 4s     
/tmp/linux-arm64-de  73%[=============>      ]  73.73M  6.88MB/s    eta 4s     
/tmp/linux-arm64-de  74%[=============>      ]  75.03M  6.78MB/s    eta 4s     
/tmp/linux-arm64-de  75%[==============>     ]  76.31M  6.44MB/s    eta 4s     
/tmp/linux-arm64-de  77%[==============>     ]  77.63M  6.40MB/s    eta 4s     
/tmp/linux-arm64-de  78%[==============>     ]  78.96M  6.26MB/s    eta 3s     
/tmp/linux-arm64-de  79%[==============>     ]  79.81M  6.00MB/s    eta 3s     
/tmp/linux-arm64-de  81%[===============>    ]  81.43M  6.10MB/s    eta 3s     
/tmp/linux-arm64-de  82%[===============>    ]  82.80M  6.16MB/s    eta 3s     
/tmp/linux-arm64-de  83%[===============>    ]  84.16M  6.25MB/s    eta 3s     
/tmp/linux-arm64-de  85%[================>   ]  85.55M  6.32MB/s    eta 2s     
/tmp/linux-arm64-de  86%[================>   ]  86.96M  6.39MB/s    eta 2s     
/tmp/linux-arm64-de  88%[================>   ]  88.42M  6.47MB/s    eta 2s     
/tmp/linux-arm64-de  89%[================>   ]  89.91M  6.54MB/s    eta 2s     
/tmp/linux-arm64-de  90%[=================>  ]  91.35M  6.60MB/s    eta 2s     
/tmp/linux-arm64-de  92%[=================>  ]  92.84M  6.67MB/s    eta 1s     
/tmp/linux-arm64-de  93%[=================>  ]  94.27M  6.75MB/s    eta 1s     
/tmp/linux-arm64-de  95%[==================> ]  95.76M  6.81MB/s    eta 1s     
/tmp/linux-arm64-de  96%[==================> ]  97.29M  6.89MB/s    eta 1s     
/tmp/linux-arm64-de  98%[==================> ]  98.85M  6.99MB/s    eta 1s     
/tmp/linux-arm64-de  99%[==================> ] 100.45M  7.07MB/s    eta 0s     
/tmp/linux-arm64-de 100%[===================>] 100.46M  7.07MB/s    in 13s     

2024-10-08 14:59:00 (7.66 MB/s) - ‘/tmp/linux-arm64-deb.deb’ saved [105335974/105335974]


Done
dpkg-deb: error: archive '/tmp/linux-arm64-deb.deb' uses unknown compression for member 'control.tar.zst', giving up
install_packages(): failed to determine a package-name for the file '/tmp/linux-arm64-deb.deb'

Failed to install Visual Studio Code!
◢◣Need help? Copy the ENTIRE terminal output or take a screenshot.
Please ask on Github: https://github.com/Botspot/pi-apps/issues/new/choose
Or on Discord: https://discord.gg/RXSTvaUvuu
[New.txt](https://github.com/user-attachments/files/17293704/New.txt)


### What are your system specs (run the following command in your terminal)?

```bash
OS: Debian GNU/Linux 11 (bullseye)
OS architecture: 64-bit
Last updated Pi-Apps on: 10/08/2024
Latest Pi-Apps version: 10/08/2024
Kernel: aarch64 6.1.21-v8+
Device model: Raspberry Pi 3 Model B Plus Rev 1.3
SOC identifier: bcm2837
Machine-id (hashed): 9e637a2ba6ee471128579dd0371d087d54504467
Serial-number (hashed): 441072864a16283af349896e13d27c064a24c131
Cpu name: Cortex-A53
Ram size: .73 GB
Raspberry Pi OS image version: 2024-07-04
Language: en_US.UTF-8
```

### (Recommended) Error log? Terminal output? Debug messages?

```bash
OS: Debian GNU/Linux 11 (bullseye)
OS architecture: 64-bit
Last updated Pi-Apps on: 10/08/2024
Latest Pi-Apps version: 10/08/2024
Kernel: aarch64 6.1.21-v8+
Device model: Raspberry Pi 3 Model B Plus Rev 1.3
SOC identifier: bcm2837
Machine-id (hashed): 9e637a2ba6ee471128579dd0371d087d54504467
Serial-number (hashed): 441072864a16283af349896e13d27c064a24c131
Cpu name: Cortex-A53
Ram size: .73 GB
Raspberry Pi OS image version: 2024-07-04
Language: en_US.UTF-8

Will install these packages: https://aka.ms/linux-arm64-deb 
Waiting until APT locks are released... Done
/tmp/linux-arm64-deb is not ending with .deb, renaming it to '/tmp/linux-arm64-deb.deb'...
Downloading linux-arm64-deb to /tmp/linux-arm64-deb.deb... 
--2024-10-08 14:58:46--  https://aka.ms/linux-arm64-deb
Resolving aka.ms (aka.ms)... 72.246.170.70
Connecting to aka.ms (aka.ms)|72.246.170.70|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://update.code.visualstudio.com/latest/linux-deb-arm64/stable [following]
--2024-10-08 14:58:47--  https://update.code.visualstudio.com/latest/linux-deb-arm64/stable
Resolving update.code.visualstudio.com (update.code.visualstudio.com)... 13.107.246.45, 2620:1ec:bdf::45
Connecting to update.code.visualstudio.com (update.code.visualstudio.com)|13.107.246.45|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://vscode.download.prss.microsoft.com/dbazure/download/stable/d78a74bcdfad14d5d3b1b782f87255d802b57511/code_1.94.0-1727877189_arm64.deb [following]
--2024-10-08 14:58:47--  https://vscode.download.prss.microsoft.com/dbazure/download/stable/d78a74bcdfad14d5d3b1b782f87255d802b57511/code_1.94.0-1727877189_arm64.deb
Resolving vscode.download.prss.microsoft.com (vscode.download.prss.microsoft.com)... 151.101.242.172, 2a04:4e42:39::684
Connecting to vscode.download.prss.microsoft.com (vscode.download.prss.microsoft.com)|151.101.242.172|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 105335974 (100M) [application/octet-stream]
Saving to: ‘/tmp/linux-arm64-deb.deb’


/tmp/linux-arm64-de   0%[                    ]       0  --.-KB/s               
/tmp/linux-arm64-de   0%[                    ] 895.21K  3.83MB/s               
/tmp/linux-arm64-de   2%[                    ]   2.78M  6.48MB/s               
/tmp/linux-arm64-de   4%[                    ]   4.69M  7.45MB/s               
/tmp/linux-arm64-de   6%[>                   ]   6.23M  7.51MB/s               
/tmp/linux-arm64-de   7%[>                   ]   7.82M  7.59MB/s               
/tmp/linux-arm64-de   9%[>                   ]   9.44M  7.67MB/s               
/tmp/linux-arm64-de  11%[=>                  ]  11.11M  7.75MB/s               
/tmp/linux-arm64-de  12%[=>                  ]  12.66M  7.75MB/s               
/tmp/linux-arm64-de  14%[=>                  ]  14.38M  7.84MB/s               
/tmp/linux-arm64-de  16%[==>                 ]  16.09M  7.91MB/s               
/tmp/linux-arm64-de  17%[==>                 ]  17.81M  7.97MB/s               
/tmp/linux-arm64-de  19%[==>                 ]  19.53M  8.02MB/s               
/tmp/linux-arm64-de  21%[===>                ]  21.31M  8.08MB/s               
/tmp/linux-arm64-de  22%[===>                ]  23.09M  8.12MB/s               
/tmp/linux-arm64-de  24%[===>                ]  24.92M  8.18MB/s    eta 9s     
/tmp/linux-arm64-de  26%[====>               ]  26.70M  8.47MB/s    eta 9s     
/tmp/linux-arm64-de  28%[====>               ]  28.33M  8.63MB/s    eta 9s     
/tmp/linux-arm64-de  29%[====>               ]  29.40M  8.16MB/s    eta 9s     
/tmp/linux-arm64-de  31%[=====>              ]  31.28M  8.28MB/s    eta 9s     
/tmp/linux-arm64-de  32%[=====>              ]  33.10M  8.34MB/s    eta 8s     
/tmp/linux-arm64-de  34%[=====>              ]  34.52M  8.30MB/s    eta 8s     
/tmp/linux-arm64-de  36%[======>             ]  36.34M  8.35MB/s    eta 8s     
/tmp/linux-arm64-de  37%[======>             ]  38.17M  8.41MB/s    eta 8s     
/tmp/linux-arm64-de  39%[======>             ]  39.97M  8.49MB/s    eta 8s     
/tmp/linux-arm64-de  41%[=======>            ]  41.79M  8.52MB/s    eta 7s     
/tmp/linux-arm64-de  43%[=======>            ]  43.59M  8.54MB/s    eta 7s     
/tmp/linux-arm64-de  45%[========>           ]  45.51M  8.60MB/s    eta 7s     
/tmp/linux-arm64-de  47%[========>           ]  47.46M  8.66MB/s    eta 7s     
/tmp/linux-arm64-de  49%[========>           ]  49.37M  8.71MB/s    eta 7s     
/tmp/linux-arm64-de  51%[=========>          ]  51.27M  8.74MB/s    eta 6s     
/tmp/linux-arm64-de  52%[=========>          ]  53.19M  8.79MB/s    eta 6s     
/tmp/linux-arm64-de  54%[=========>          ]  55.10M  8.83MB/s    eta 6s     
/tmp/linux-arm64-de  56%[==========>         ]  57.03M  9.27MB/s    eta 6s     
/tmp/linux-arm64-de  58%[==========>         ]  58.77M  9.08MB/s    eta 6s     
/tmp/linux-arm64-de  59%[==========>         ]  60.19M  8.98MB/s    eta 5s     
/tmp/linux-arm64-de  61%[===========>        ]  61.65M  8.98MB/s    eta 5s     
/tmp/linux-arm64-de  62%[===========>        ]  62.94M  8.83MB/s    eta 5s     
/tmp/linux-arm64-de  63%[===========>        ]  64.05M  8.61MB/s    eta 5s     
/tmp/linux-arm64-de  64%[===========>        ]  65.20M  8.37MB/s    eta 5s     
/tmp/linux-arm64-de  66%[============>       ]  66.37M  8.18MB/s    eta 4s     
/tmp/linux-arm64-de  67%[============>       ]  67.54M  7.97MB/s    eta 4s     
/tmp/linux-arm64-de  68%[============>       ]  68.77M  7.73MB/s    eta 4s     
/tmp/linux-arm64-de  69%[============>       ]  69.99M  7.54MB/s    eta 4s     
/tmp/linux-arm64-de  70%[=============>      ]  71.23M  7.36MB/s    eta 4s     
/tmp/linux-arm64-de  72%[=============>      ]  72.47M  7.06MB/s    eta 4s     
/tmp/linux-arm64-de  73%[=============>      ]  73.73M  6.88MB/s    eta 4s     
/tmp/linux-arm64-de  74%[=============>      ]  75.03M  6.78MB/s    eta 4s     
/tmp/linux-arm64-de  75%[==============>     ]  76.31M  6.44MB/s    eta 4s     
/tmp/linux-arm64-de  77%[==============>     ]  77.63M  6.40MB/s    eta 4s     
/tmp/linux-arm64-de  78%[==============>     ]  78.96M  6.26MB/s    eta 3s     
/tmp/linux-arm64-de  79%[==============>     ]  79.81M  6.00MB/s    eta 3s     
/tmp/linux-arm64-de  81%[===============>    ]  81.43M  6.10MB/s    eta 3s     
/tmp/linux-arm64-de  82%[===============>    ]  82.80M  6.16MB/s    eta 3s     
/tmp/linux-arm64-de  83%[===============>    ]  84.16M  6.25MB/s    eta 3s     
/tmp/linux-arm64-de  85%[================>   ]  85.55M  6.32MB/s    eta 2s     
/tmp/linux-arm64-de  86%[================>   ]  86.96M  6.39MB/s    eta 2s     
/tmp/linux-arm64-de  88%[================>   ]  88.42M  6.47MB/s    eta 2s     
/tmp/linux-arm64-de  89%[================>   ]  89.91M  6.54MB/s    eta 2s     
/tmp/linux-arm64-de  90%[=================>  ]  91.35M  6.60MB/s    eta 2s     
/tmp/linux-arm64-de  92%[=================>  ]  92.84M  6.67MB/s    eta 1s     
/tmp/linux-arm64-de  93%[=================>  ]  94.27M  6.75MB/s    eta 1s     
/tmp/linux-arm64-de  95%[==================> ]  95.76M  6.81MB/s    eta 1s     
/tmp/linux-arm64-de  96%[==================> ]  97.29M  6.89MB/s    eta 1s     
/tmp/linux-arm64-de  98%[==================> ]  98.85M  6.99MB/s    eta 1s     
/tmp/linux-arm64-de  99%[==================> ] 100.45M  7.07MB/s    eta 0s     
/tmp/linux-arm64-de 100%[===================>] 100.46M  7.07MB/s    in 13s     

2024-10-08 14:59:00 (7.66 MB/s) - ‘/tmp/linux-arm64-deb.deb’ saved [105335974/105335974]


Done
dpkg-deb: error: archive '/tmp/linux-arm64-deb.deb' uses unknown compression for member 'control.tar.zst', giving up
install_packages(): failed to determine a package-name for the file '/tmp/linux-arm64-deb.deb'

Failed to install Visual Studio Code!
◢◣Need help? Copy the ENTIRE terminal output or take a screenshot.
Please ask on Github: https://github.com/Botspot/pi-apps/issues/new/choose
Or on Discord: https://discord.gg/RXSTvaUvuu
```

**Comment by theofficialgman** [comment]
opened upstream bug https://github.com/microsoft/vscode/issues/230825

**Comment by theofficialgman** [comment]
Completed in upstream main but not yet in a release. https://github.com/microsoft/vscode/commit/ba6907515e1423671b91deb96b2088c5cd8d35a7

Keeping bug open until https://github.com/microsoft/vscode/issues/230418 is resolved (the original report)

**Comment by theofficialgman** [comment]
Issue resolved upstream with 1.94.2 release.
